router/router.go

```diff
@@ -0,0 +1,208 @@
+package router
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pokt-foundation/transaction-db/types"
+	jsonresponse "github.com/pokt-foundation/utils-go/json-response"
+
+	"github.com/sirupsen/logrus"
+)
+
+type driver interface {
+	WriteError(ctx context.Context, poktErr types.Error) error
+	WriteSession(ctx context.Context, session types.PocketSession) error
+	WriteRegion(ctx context.Context, region types.PortalRegion) error
+	WriteRelay(ctx context.Context, relay types.Relay) error
+	ReadRelay(ctx context.Context, relayID int) (types.Relay, error)
+}
+
+type Router struct {
+	Router  *mux.Router
+	Driver  driver
+	APIKeys map[string]bool
+	log     *logrus.Logger
+}
+
+func (rt *Router) logError(err error) {
+	fields := logrus.Fields{
+		"err": err.Error(),
+	}
+
+	rt.log.WithFields(fields).Error(err)
+}
+
+// NewRouter returns router instance
+func NewRouter(driver driver, apiKeys map[string]bool, logger *logrus.Logger) (*Router, error) {
+	rt := &Router{
+		Driver:  driver,
+		Router:  mux.NewRouter(),
+		APIKeys: apiKeys,
+		log:     logger,
+	}
+
+	rt.Router.HandleFunc("/", rt.HealthCheck).Methods(http.MethodGet)
+
+	rt.Router.HandleFunc("/v0/error", rt.CreateError).Methods(http.MethodPost)
+	rt.Router.HandleFunc("/v0/session", rt.CreateSession).Methods(http.MethodPost)
+	rt.Router.HandleFunc("/v0/region", rt.CreateRegion).Methods(http.MethodPost)
+	rt.Router.HandleFunc("/v0/relay", rt.CreateRelay).Methods(http.MethodPost)
+	rt.Router.HandleFunc("/v0/relay/{id}", rt.GetRelay).Methods(http.MethodGet)
+
+	rt.Router.Use(rt.AuthorizationHandler)
+
+	return rt, nil
+}
+
+func (rt *Router) AuthorizationHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// This is the path of the health check endpoint
+		if r.URL.Path == "/" {
+			h.ServeHTTP(w, r)
+
+			return
+		}
+
+		if !rt.APIKeys[r.Header.Get("Authorization")] {
+			w.WriteHeader(http.StatusUnauthorized)
+			_, err := w.Write([]byte("Unauthorized"))
+			if err != nil {
+				panic(err)
+			}
+
+			return
+		}
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+func (rt *Router) HealthCheck(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	_, err := w.Write([]byte("Transaction HTTP DB is up and running!"))
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (rt *Router) CreateError(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	decoder := json.NewDecoder(r.Body)
+
+	var errToAdd types.Error
+	err := decoder.Decode(&errToAdd)
+	if err != nil {
```

Comment:
Just a nitpick for this and other instances that follow this pattern: why do you declare the variable outside when you're only using it within the if logic? As you could declare it on the same line as the if